src/gadgets/ecc.rs

```diff
@@ -160,6 +161,84 @@ impl<C: CurveAffine<Base = F>, F: PrimeFieldBits, const T: usize> EccChip<C, F,
         Ok(acc)
     }
 
+    // optimization version for scalar_mul
+    // we assume the point p0 is not infinity here
+    pub fn scalar_mul_opt(
+        &self,
+        ctx: &mut RegionCtx<'_, C::Base>,
+        p0: &AssignedPoint<C>,
+        scalar_bits: &[AssignedValue<C::Base>],
+    ) -> Result<AssignedPoint<C>, Error> {
+        if let Some((x, y)) = p0.coordinates_values() {
+            if x == C::Base::ZERO && y == C::Base::ZERO {
+                return Err(Error::Synthesis);
```

Comment:
Please add log message for this case, halo2 errors too common

Reply:
added debug!

Reply:
This is not `debug!`, this is `error!`

The halo2 error design is bad, not including context, debugging a real working system on a prod we will get a synthesis error and not know what to do about it. That's why in such cases we "at least" add a error log here